core/src/main/java/com/opensymphony/xwork2/ognl/OgnlUtil.java

```diff
@@ -184,6 +184,19 @@ protected void setDisallowProxyMemberAccess(String disallowProxyMemberAccess) {
         this.disallowProxyMemberAccess = Boolean.parseBoolean(disallowProxyMemberAccess);
     }
 
+    /**
+     * @param maxLength Injects the Struts OGNL maximum expression length.
+     */
+    @Inject(value = StrutsConstants.STRUTS_OGNL_EXPRESSION_MAX_LENGTH, required = false)
+    protected void applyExpressionMaxLength(String maxLength) {
```

Comment:
Since `applyExpressionMaxLength()` may produce an exception, maybe the set attempts should be guarded:
```
try {
...
} catch (Exception ex) {
    LOG.warn("Unable to set OGNL Expression Max Length {}.", maxLength, ex);
    // Decide whether to only log exception or also re-throw with: "throw ex;"
}
```

Reply:
AFAIK Struts didn't use to operate like that. It fails in such conditions with fatal BadConfException and compels user to fix it.

Reply:
Hi Yasser.

I don't think Struts configuration normally operates like that, but OGNL behaviour around the applyExpressionMaxLength could change in the future.  The log warning would make it clearer what failed either way (from a Struts perspective, including what maximum length was being requested when it failed) ... and the exception can still be re-thrown to compel the user to fix it.

The "controlled warning" seemed like it might be of some use, even if the exception is still re-thrown.  :smile: